controllers/article.js

```diff
@@ -86,48 +94,77 @@ export default {
 
     return res.status(201).send({ message: 'Your report has been registered' });
   },
-  async getArticleByCategory(req, res) {
-    const { category } = req.query;
 
+  async getArticles(req, res) {
+    const {
+      category,
+      author,
+      tag,
+      keywords
+    } = req.query;
+    const queryArray = [
+      {
+        [or]: {
+          title: { [iLike]: keywords ? `%${keywords}%` : '%' },
+          description: { [iLike]: keywords ? `%${keywords}%` : '%' },
+          body: { [iLike]: keywords ? `%${keywords}%` : '%' }
+        },
+      },
+      {
+        category: category || categories
+      },
+      {
+        tags: tag ? { [contains]: [tag] } : { [notIn]: [] }
+      }
+    ];
+    if (author) {
+      queryArray.push({
+        [or]: {
+          '$User.Profile.lastName$': {
+            [iLike]: `%${author}%`
+          },
+          '$User.Profile.firstName$': {
+            [iLike]: `%${author}%`
+          },
+        }
+
+      });
+    }
     const article = await Article.findAll({
```

Comment:
Help change this to `articles` to appropriately connote multiple results expected.